stdlib/public/core/Collection.swift

```diff
@@ -584,19 +584,17 @@ extension SequenceType
   // A fast implementation for when you are backed by a contiguous array.
   public func _initializeTo(ptr: UnsafeMutablePointer<Generator.Element>)
     -> UnsafeMutablePointer<Generator.Element> {
-    let s = self._baseAddressIfContiguous
-    if s != nil {
-      let count = self.count
-      ptr.initializeFrom(s, count: count)
-      _fixLifetime(self._owner)
-      return ptr + count
-    } else {
+    guard let s = self._baseAddressIfContiguous else {
       var p = ptr
       for x in self {
         p++.initialize(x)
       }
       return p
     }
```

Comment:
This has been suggested already.  `guard` reads well when used for quick early exits.  There is a lot of code in the `guard` here.